### PR TITLE
Misc builder script fixes & refactors

### PR DESCRIPTION
**Pull request by palpatim**
- Consolidate requirements to top-level
- Add sts:GetParametersByPath to CircleCIExecutionRole
- Minor refactor of config builder
- Moved builder into 'scripts' subdirectory with its own requirements.txt

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/integ_test_resources/common/scripts/test/device_config_builder_test.py [hunk 1]
```diff
@@ -68,6 +68,26 @@ def test_build_package_data_with_nesting(self):
         )
 
 
+    def test_get_credential_data(self):
```
Comment (jpignata) [inline src/integ_test_resources/common/scripts/test/device_config_builder_test.py:71]: I might put some time into ensuring we don't litter a test run's environment by either [patching](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch.dict) `os.environ` or ensuring these keys  are returned to their original state. It doesn't matter with the current tests, but could matter in the future.

File: requirements.txt [hunk 2]
```diff
@@ -1 +1,13 @@
...
+aws_cdk.aws_kinesisfirehose
+aws_cdk.aws_kms
+aws_cdk.aws_logs
+aws_cdk.aws_pinpoint
+aws-cdk.aws_s3
+aws_cdk.aws_ssm
+aws_cdk.core
 black
+boto3
+isort
```
Comment (jpignata) [inline requirements.txt:13]: Is this required because we have a pre-commit hook or something?

**Comment by jpignata** [comment]
LGTM!

File: requirements.txt (hunk 2 above)
Comment (palpatim) [reply requirements.txt:13]: No precommit hook yet, but I'm planning on doing a reformat and updating the contribution guide with code style guidelines

File: src/integ_test_resources/common/scripts/test/device_config_builder_test.py (hunk 1 above)
Comment (palpatim) [reply src/integ_test_resources/common/scripts/test/device_config_builder_test.py:71]: That's a really nice callout, thanks! Will add in an upcoming PR

**Pull request merged**